src/track/serato/markers2.h

```diff
@@ -92,102 +113,103 @@ class SeratoMarkers2BpmlockEntry : public SeratoMarkers2Entry {
 
     quint32 length() const override;
 
-private:
+  private:
     bool m_locked;
 };
 
-inline
-bool operator==(const SeratoMarkers2BpmlockEntry& lhs,
-                const SeratoMarkers2BpmlockEntry& rhs) {
+inline bool operator==(const SeratoMarkers2BpmlockEntry& lhs,
+        const SeratoMarkers2BpmlockEntry& rhs) {
     return (lhs.isLocked() == rhs.isLocked());
 }
 
-inline
-bool operator!=(const SeratoMarkers2BpmlockEntry& lhs,
-                const SeratoMarkers2BpmlockEntry& rhs) {
+inline bool operator!=(const SeratoMarkers2BpmlockEntry& lhs,
+        const SeratoMarkers2BpmlockEntry& rhs) {
     return !(lhs == rhs);
 }
 
-inline
-QDebug operator<<(QDebug dbg, const SeratoMarkers2BpmlockEntry& arg) {
+inline QDebug operator<<(QDebug dbg, const SeratoMarkers2BpmlockEntry& arg) {
     return dbg << "locked =" << arg.isLocked();
 }
 
 class SeratoMarkers2ColorEntry : public SeratoMarkers2Entry {
-public:
-    SeratoMarkers2ColorEntry(QColor color)
-        : m_color(color) {
+  public:
+    SeratoMarkers2ColorEntry(QRgb colorMask)
+            : m_colorMask(colorMask) {
     }
 
     SeratoMarkers2ColorEntry()
-        : m_color(QColor()) {
+            : m_colorMask(kDefaultTrackColorMask) {
```

Comment:
Why is this a "Mask"? A Mask is for me a bit field that is applied like this
maskedValue = Value & Mask;
maskedValue = Value | ~Mask;

Reply:
I initially thought it was indeed some kind of mask, but now I realized it's actually the color displayed in the track color picker, which is different from the color displayed in the color column of the library view:

| Color Picker / Stored in Tag                                       | Show in Library Column
| ------------------------------------------------------------------ | ------------------------------------------------------------------
| ![FF99FF](http://dummyimage.com/20x20/FF99FF/FF99FF.gif) `#FF99FF` | ![993399](http://dummyimage.com/20x20/993399/993399.gif) `#993399`
| ![FF99DD](http://dummyimage.com/20x20/FF99DD/FF99DD.gif) `#FF99DD` | ![993377](http://dummyimage.com/20x20/993377/993377.gif) `#993377`
| ![FF99BB](http://dummyimage.com/20x20/FF99BB/FF99BB.gif) `#FF99BB` | ![993355](http://dummyimage.com/20x20/993355/993355.gif) `#993355`
| ![FF9999](http://dummyimage.com/20x20/FF9999/FF9999.gif) `#FF9999` | ![993333](http://dummyimage.com/20x20/993333/993333.gif) `#993333`
| ![FFBB99](http://dummyimage.com/20x20/FFBB99/FFBB99.gif) `#FFBB99` | ![995533](http://dummyimage.com/20x20/995533/995533.gif) `#995533`
| ![FFDD99](http://dummyimage.com/20x20/FFDD99/FFDD99.gif) `#FFDD99` | ![997733](http://dummyimage.com/20x20/997733/997733.gif) `#997733`
| ![FFFF99](http://dummyimage.com/20x20/FFFF99/FFFF99.gif) `#FFFF99` | ![999933](http://dummyimage.com/20x20/999933/999933.gif) `#999933`
| ![DDFF99](http://dummyimage.com/20x20/DDFF99/DDFF99.gif) `#DDFF99` | ![779933](http://dummyimage.com/20x20/779933/779933.gif) `#779933`
| ![BBFF99](http://dummyimage.com/20x20/BBFF99/BBFF99.gif) `#BBFF99` | ![559933](http://dummyimage.com/20x20/559933/559933.gif) `#559933`
| ![99FF99](http://dummyimage.com/20x20/99FF99/99FF99.gif) `#99FF99` | ![339933](http://dummyimage.com/20x20/339933/339933.gif) `#339933`
| ![99FFBB](http://dummyimage.com/20x20/99FFBB/99FFBB.gif) `#99FFBB` | ![339955](http://dummyimage.com/20x20/339955/339955.gif) `#339955`
| ![99FFDD](http://dummyimage.com/20x20/99FFDD/99FFDD.gif) `#99FFDD` | ![339977](http://dummyimage.com/20x20/339977/339977.gif) `#339977`
| ![99FFFF](http://dummyimage.com/20x20/99FFFF/99FFFF.gif) `#99FFFF` | ![339999](http://dummyimage.com/20x20/339999/339999.gif) `#339999`
| ![99DDFF](http://dummyimage.com/20x20/99DDFF/99DDFF.gif) `#99DDFF` | ![337799](http://dummyimage.com/20x20/337799/337799.gif) `#337799`
| ![99BBFF](http://dummyimage.com/20x20/99BBFF/99BBFF.gif) `#99BBFF` | ![335599](http://dummyimage.com/20x20/335599/335599.gif) `#335599`
| ![9999FF](http://dummyimage.com/20x20/9999FF/9999FF.gif) `#9999FF` | ![333399](http://dummyimage.com/20x20/333399/333399.gif) `#333399`
| ![BB99FF](http://dummyimage.com/20x20/BB99FF/BB99FF.gif) `#BB99FF` | ![553399](http://dummyimage.com/20x20/553399/553399.gif) `#553399`
| ![DD99FF](http://dummyimage.com/20x20/DD99FF/DD99FF.gif) `#DD99FF` | ![773399](http://dummyimage.com/20x20/773399/773399.gif) `#773399`
| ![FFFFFF](http://dummyimage.com/20x20/FFFFFF/FFFFFF.gif) `#FFFFFF` | ![333333](http://dummyimage.com/20x20/333333/333333.gif) `#333333`
| ![BBBBBB](http://dummyimage.com/20x20/BBBBBB/BBBBBB.gif) `#BBBBBB` | ![555555](http://dummyimage.com/20x20/555555/555555.gif) `#555555`
| ![999999](http://dummyimage.com/20x20/999999/999999.gif) `#999999` | ![090909](http://dummyimage.com/20x20/090909/090909.gif) `#090909`

How exactly they are mapped. The last 3 entries destroy the pattern that you just substract 0x666666 ;-)

Reply:
Ok, I got it: https://github.com/Holzhaus/serato-tags/blob/master/docs/colors.md#track-colors

Reply:
I renamed it back to `trackColor`. The actual conversion from stored value to column value willl be added to the import code, not here. It also applies to the "Serato Markers2" tag.